### PR TITLE
Fix gh action

### LINKED ISSUE #3723: Fix GH Action that adds reviewers
It is failing with `Error: HttpError: Resource not accessible by integration`: https://github.com/vmware-tanzu/velero/runs/2410566688?check_suite_focus=true

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

### PR DESCRIPTION
**Pull request by carlisia**
The `number_of_reviewers` option was at the wrong level. Hopefully this will be "the fix".

Signed-off-by: Carlisia <carlisia@grokkingtech.io>

# Does your change fix a particular issue?

Fixes #3723

# Please indicate you've done the following:

- [x] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [x] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required`.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Comment by carlisia** [comment]
This is still failing: https://github.com/vmware-tanzu/velero/runs/2453464772?check_suite_focus=true

I made some fixes but it is the added permission for the job that makes me hopeful this will work once merged.

If it does not work, my next thing to look at would be to create a separate secret token with the proper repo level permissions.

**Comment by carlisia** [comment]
I have figured this out. The `secrets.GITHUB_TOKEN` will only permit "read" access on forks for the "pull_request" GH action trigger. Any permission added to a GH action configuration (like I did in this PR) will only ever pertain to the main branch.

Possible solutions: 
- create a personal token with the proper repo level scope for writing, and use that in the action
- change the action trigger from `pull_request` to `pull_request_target`. This one will open up the permission on the fork.

I am still investigating some security implications (https://securitylab.github.com/research/github-actions-preventing-pwn-requests/) to make sure we go with the most secure option. I'm also tempted to include a blurb in our maintainer's doc so everyone is aware of the potential for exploitations due to misconfiguring GH actions.

**Comment by carlisia** [comment]
tl;dr: regardless of what the GH action documents, always use `pull_request_target` + `secrets.GITHUB_TOKEN` for workflows that make change to PRs such as adding labels, authors, reviewers, etc (as opposed to running code from the PR for example). Use that instead of `pull_request`, because using that with `secrets.GITHUB_TOKEN` we only get "read" access on the fork, and would so require using a personal token with the write perms to the repo, which is not ideal.

**Review by zubron — APPROVED** [review]
Thanks for your efforts investigating this! So using `pull_request_target` will solve the permission issues? We can merge this now and investigate further if needed.

**Comment by carlisia** [comment]
> So using pull_request_target will solve the permission issues?
Yes. That's why they have both `pull_request` and `pull_request_target`. The first will not give write perms on the fork and is to be used for actions that have the potential to cause security issues. The second is meant to give write perms.

**Comment by zubron** [comment]
@carlisia Awesome, thanks for the explanation :)

**Review by jenting — APPROVED** [review]
Let's let the ball rolling.
